src/runtime_src/core/pcie/driver/linux/xocl/subdev/ert_30.c

```diff
@@ -697,23 +782,32 @@ static inline int process_ert_rq(struct xocl_ert_30 *ert_30)
 	u32 slot_addr = 0, i;
 	struct ert_packet *epkt = NULL;
 	xdev_handle_t xdev = xocl_get_xdev(ert_30->pdev);
+	struct kds_client *ev_client = NULL;
 
 	if (!ert_30->num_rq)
 		return 0;
 
+	ev_client = first_event_client_or_null(ert_30);
 	list_for_each_entry_safe(ecmd, next, &ert_30->rq, list) {
+		struct kds_command *xcmd = ecmd->xcmd;
+
+		if (unlikely(ert_30->bad_state || (ev_client == xcmd->client))) {
+			ERTUSER_ERR(ert_30, "%s abort\n", __func__);
+			ecmd->status = KDS_ABORT;
+			list_move_tail(&ecmd->list, &ert_30->cq);
+			--ert_30->num_rq;
+			++ert_30->num_cq;
+			continue;
+		}
 
 		if (cmd_opcode(ecmd) == OP_CONFIG) {
 			if (ert_cfg_cmd(ert_30, ecmd)) {
-				struct kds_command *xcmd;
 
 				ERTUSER_ERR(ert_30, "%s config cmd error\n", __func__);
-				list_del(&ecmd->list);
-				xcmd = ecmd->xcmd;
-				xcmd->cb.notify_host(xcmd, KDS_ABORT);
-				xcmd->cb.free(xcmd);
-				ert_30_free_cmd(ecmd);
+				ecmd->status = KDS_ABORT;
```

Comment:
I just aware that we return KDS_ABORT when ert_cfg_cmd() return error. Should we use KDS_ERROR?